pkg/collector/collector.go

```diff
@@ -122,66 +124,33 @@ func (c *Collector) RunCheck(ch check.Check) (check.ID, error) {
 	return ch.ID(), nil
 }
 
-// ReloadCheck stops and restart a check with a new configuration
-func (c *Collector) ReloadCheck(id check.ID, config, initConfig integration.Data, newSource string) error {
+// StopCheck halts a check and remove the instance
+func (c *Collector) StopCheck(id check.ID) error {
 	if !c.started() {
 		return fmt.Errorf("the collector is not running")
 	}
 
-	// do we know this check instance?
-	// BUG(massi): we could create the Check if it doesn't exist, see https://github.com/DataDog/datadog-agent/pull/148
-	// for reference
-	if !c.find(id) {
+	ch, found := c.get(id)
+	if !found {
 		return fmt.Errorf("cannot find a check with ID %s", id)
 	}
 
-	c.m.Lock()
-	defer c.m.Unlock()
-
 	// unschedule the instance
 	err := c.scheduler.Cancel(id)
 	if err != nil {
 		return fmt.Errorf("an error occurred while canceling the check schedule: %s", err)
 	}
 
-	// stop the instance
 	err = c.runner.StopCheck(id)
 	if err != nil {
+		// still attempt to cancel the check before returning the error
+		_ = c.cancelCheck(ch, cancelCheckTimeout)
 		return fmt.Errorf("an error occurred while stopping the check: %s", err)
 	}
 
-	// re-configure
-	check := c.checks[id]
-	err = check.Configure(config, initConfig, newSource)
+	err = c.cancelCheck(ch, cancelCheckTimeout)
 	if err != nil {
-		return fmt.Errorf("error configuring the check with ID %s", id)
-	}
-
-	// re-schedule
-	c.scheduler.Enter(check) //nolint:errcheck
-
-	return nil
-}
-
-// StopCheck halts a check and remove the instance
-func (c *Collector) StopCheck(id check.ID) error {
-	if !c.started() {
-		return fmt.Errorf("the collector is not running")
-	}
-
-	if !c.find(id) {
-		return fmt.Errorf("cannot find a check with ID %s", id)
-	}
-
-	// unschedule the instance
-	err := c.scheduler.Cancel(id)
-	if err != nil {
-		return fmt.Errorf("an error occurred while canceling the check schedule: %s", err)
-	}
-
-	err = c.runner.StopCheck(id)
-	if err != nil {
-		return fmt.Errorf("an error occurred while stopping the check: %s", err)
+		return fmt.Errorf("an error occurred while calling check.Cancel(): %s", err)
```

Comment:
would be nice to print the check ID

Reply:
The errors returned by `c.runner.StopCheck` and `c.cancelCheck` already always include the check ID in them, so it would redundant. I don't have a strong opinion though.

Reply:
fair enough